WebDevStudios/wds-bt
# #36 Retain existing CSS classes on buttons on size change

# Closes

[WDSBT-63](https://webdevstudios.atlassian.net/browse/WDSBT-63)
https://github.com/WebDevStudios/wds-bt/issues/36

## Description

Updated the way the class name for the custom button size control is handled.

Took inspiration from how Gutenberg handles this for block styles with `replaceActiveStyle()`, but the gist is that it will create a token list from the existing class names on the block and then removes or adds a class name when appropriate and returns the new value. 

So, the current class name will be removed from the list if one exists and a new class name will be added if a size was selected.

## Mobile & Desktop Screenshots/Recordings

![fixed-buttons](https://github.com/user-attachments/assets/382171f9-275f-4731-9ecf-cbbd1156d5b0)

## Added to documentation?


## Added tests?


## Testing Instructions

Confirm that you can change the style and size on a button block independently without wiping out one or the other.

-----

## Reviewer's Testing Checklist



As a reviewer, please verify that the relevant testing criteria are fulfilled and confirmed before approving this Pull Request.


## [optional] Additional Reviewer Notes or Considerations?


[WDSBT-63]: https://webdevstudios.atlassian.net/browse/WDSBT-63?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
